### ISSUE TITLE
Generate wildcards for generic type

### ISSUE THREAD

**Issue by jakoss** [mars885/hilt-binder#7]
I want to create multibinding for such collection:
```kotlin
interface HiltMavericksEntryPoint {
    val viewModelFactories: Map<Class<out MavericksViewModel<*>>, AssistedViewModelFactory<*, *>>
}
```
I'm trying to use hilt binder like that:
```kotlin
@BindType(
        to = AssistedViewModelFactory::class,
        contributesTo = BindType.Collection.MAP
    )
    @ViewModelKey(DashboardViewModel::class)
    @AssistedFactory
    interface Factory : AssistedViewModelFactory<DashboardViewModel, DashboardState> {
        override fun create(state: DashboardState): DashboardViewModel
    }
```

this generated code like that:
```java
@Module
@InstallIn(SingletonComponent.class)
public interface TestModule {
    @Binds
    @IntoMap
    @ViewModelKey(DashboardViewModel.class)
    AssistedViewModelFactory bind(
            DashboardViewModel.Factory binding);
}
```
but dagger won't accept that as a member of `Map<Class<out MavericksViewModel<*>>, AssistedViewModelFactory<*, *>>`. I tested very similar module and it's working fine:
```java
@Module
@InstallIn(SingletonComponent.class)
public interface TestModule {
    @Binds
    @IntoMap
    @ViewModelKey(DashboardViewModel.class)
    AssistedViewModelFactory<?, ?> bind(
            DashboardViewModel.Factory binding);
}
```

The only thing i added are the wildcards for the generic type, and now it's working great.

I even tried to workaround this by using `@JvmSuppressWildcards`, but does not seem to help. Even this weird construct is failing:
```kotlin
@JvmSuppressWildcards
    val viewModelFactories: @JvmSuppressWildcards Map<@JvmSuppressWildcards Class<out @JvmSuppressWildcards MavericksViewModel<*>>, @JvmSuppressWildcards AssistedViewModelFactory<*, *>>
```

Is is possible to use hilt-binder multibinding with generic types?

**Comment by mars885** [comment]
The current version has some issues with generics that I'd like to fix and improve in the upcoming release. Your problem is actually one of those issues.

Right now, you don't have much of a choice other than specifying the bindings with wildcards yourself.

**Comment by jakoss** [comment]
I guess i'll have to wait a little longer then :)

**Comment by mars885** [comment]
Your problem has been addressed in the new `1.0.0-alpha02` version. Take a look at the [changelog](https://github.com/mars885/hilt-binder/blob/master/CHANGELOG.md#version-100-alpha02-2021-04-25) to see what changed.

**Comment by jakoss** [comment]
Works perfectly, thank you!
